core/src/HIP/Kokkos_HIP_Abort.hpp

```diff
@@ -57,14 +57,8 @@ namespace Impl {
 // directive to the optimizer.
 [[noreturn]] __device__ __attribute__((noinline)) inline void hip_abort(
     char const *msg) {
-#ifdef NDEBUG
-  (void)msg;
-#else
-  // disable printf on release builds, as it has a non-trivial performance
-  // impact
-  printf("Aborting with message `%s'.\n", msg);
-#endif
-  abort();
+  const char empty[] = "";
+  __assert_fail(msg, empty, 0, empty);
```

Comment:
Shouldn't this be guarded by a check for `HIP_VERSION` since we only require 4.5.0?

Reply:
HIP version was the same for ROCm 5.0.0 and 5.0.2 on Crusher